goth_tests/domain/ya-provider/test_provider_multi_activity.py

```diff
@@ -0,0 +1,98 @@
+"""End to end tests for requesting WASM tasks using goth REST API clients."""
+
+import json
+import logging
+from pathlib import Path
+from typing import List, Optional
+
+import pytest
+
+from goth.address import (
+    PROXY_HOST,
+    YAGNA_REST_URL,
+)
+from goth.configuration import load_yaml, Override
+from goth.node import node_environment
+from goth.runner import Runner
+from goth.runner.probe import ProviderProbe, RequestorProbe
+
+from goth_tests.helpers.activity import wasi_exe_script
+from goth_tests.helpers.negotiation import negotiate_agreements, DemandBuilder
+from goth_tests.helpers.payment import pay_all
+
+logger = logging.getLogger("goth.test.multi-activity")
+
+
+@pytest.mark.asyncio
+async def test_provider_multi_activity(
+    common_assets: Path,
+    config_overrides: Optional[List[Override]],
+    log_dir: Path,
+):
+    """Test provider handling multiple activities in single Agreement."""
+
+    nodes = [
+        {"name": "requestor", "type": "Requestor"},
+        {"name": "provider-1", "type": "VM-Wasm-Provider", "use-proxy": True},
+    ]
+    config_overrides = config_overrides or []
+    config_overrides.append(("nodes", nodes))
```

Comment:
Wouldn't it be better to always have a list in `config_overrides`, never `None`?

Reply:
That makes sense, updated in 889aeac7a7772c25eb434e06cf4d63273e8844ec